### DISCUSSION TITLE
Using brew-managed python3 instead of system python3

### DISCUSSION THREAD

**Discussion by j-hui** [qutebrowser/qutebrowser#5706] [Q&A] [answered]
I'm using macOS 10.15.6 with Qutebrowser version 1.13.1.

I'm currently trying to use a userscript that depends on the Pyperclip module, which I've installed using `pip3` (installed using Homebrew). I can import this dependency using my Homebrew-installed Python executable (`/usr/local/opt/python@3.8/bin/python3.8`), but Qutebrowser isn't able to import this executable because it is using the system-provided Python3 executable (`/Library/Developer/CommandLineTools/usr/bin/python3`).

Is there any way to configure Qutebrowser to use the Homebrew-managed Python environment?

**Comment by The-Compiler** [comment, marked as answer]
Userscripts are run as external executables, not (or at least not necessarily) with the Python interpreter qutebrowser itself uses (which is bundled with it, not the system-wide one).

If you run the userscript manually from the terminal, you'll see the same behavior - either make sure the brew Python is in the `$PATH` before the system-wide one, or adjust the [shebang](https://en.wikipedia.org/wiki/Shebang_(Unix)) so that it poins to the brew Python.

**Reply by j-hui** [reply]
I'm reluctant to adjust the shebang since that would entail customizing each userscript on a per-environment basis. It's already as it should be --- `#!/usr/bin/env python3` --- and I'd like to adjust my environment to use the right `PATH`.

I don't see the same behavior in the terminal since I already adjust my `PATH` using my `.bashrc`. I'm not sure how Qutebrowser obtains its `PATH`, but Python tells me that it is: `/Applications/qutebrowser.app/Contents/MacOS:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/usr/local/bin`. It's picking up `python3` from `/usr/bin`.

What is the recommended workaround? Fwiw I installed Qutebrowser using `brew cask`.

**Reply by toofar** [reply]
See the "Unable to use spawn on MacOS" troubleshooting entry in the faq https://qutebrowser.org/doc/faq.html
You can apparently also set the path you want for all applications via launchd https://github.com/qutebrowser/qutebrowser/issues/4273#issuecomment-595892960
You may also be able to set os.env['PATH'] in config.py but I'm not sure if that would work (I seem to remember a comment about it but can't find it).
